### PR TITLE
Use spot instances for dask workers

### PR DESCRIPTION
**Pull request by yuvipanda**
- Don't change the instance type of the dask workers - so no
  mixed instances. This is a straight port of what we have now
  to just use spot instances.
- x1 spot instances can't be created due to quota issues,
  so we remove them from the pool.

Fixes https://github.com/2i2c-org/pilot-hubs/issues/490

### CODE REVIEW

**Comment by yuvipanda** [comment]
I've deployed and tested this.

**Comment by yuvipanda** [comment]
Issues to create from here:

- Fix x1 instances, for both notebooks and dask
- Consider using mixed-instance types, to ensure better pricing & availability of nodes

**Comment by choldgraf** [comment]
@yuvipanda I've created a tracking issue for this one, think we could use that to track the extra items you wanted to follow-up on? And merge this PR since it is deployed? https://github.com/2i2c-org/pilot-hubs/issues/607

File: eksctl/carbonplan.jsonnet [hunk 1]
```diff
@@ -19,6 +18,15 @@ local nodes = [
     { instanceType: "x1.32xlarge" }
 ];
 
+// Node definitions for notebook nodes. Only `instanceType` is
+// supported as a property.
```
Comment (consideRatio) [inline eksctl/carbonplan.jsonnet:22]:
```suggestion
// Node definitions for dask worker nodes. Only `instanceType` is
// supported as a property.
```

File: eksctl/carbonplan.jsonnet [hunk 2]
```diff
@@ -60,12 +69,12 @@ cluster {
                 "hub.jupyter.org/dedicated": "user:NoSchedule"
             },
 
-        } + n for n in nodes
+        } + n for n in notebookNodes
```
Comment (consideRatio) [inline eksctl/carbonplan.jsonnet:72]: I don't know jsonnet, but there is an inconsistency here.

Below it sais `} for n in daskNodes`, but here it sais `} + n for n in notebookNodes`. I'll go ahead and guess this is wrong, but perhaps its the other that is wrong? No clue.

```suggestion
        } n for n in notebookNodes
```

File: eksctl/libsonnet/nodegroup.jsonnet [hunk 3]
```diff
@@ -15,12 +15,12 @@ local makeCloudTaints(taints) = {
   availabilityZones: [],
   minSize: 0,
   desiredCapacity: self.minSize,
-  instanceType: '',
   volumeSize: 80,
   labels+: {
 	  // Add instance type as label to nodegroups, so they
-	  // can be picked up by the autoscaler
-    'node.kubernetes.io/instance-type': $.instanceType,
+	  // can be picked up by the autoscaler. If using spot instances,
```
Comment (consideRatio) [inline eksctl/libsonnet/nodegroup.jsonnet:21]:
```suggestion
    // can be picked up by the autoscaler. If using spot instances,
```

Indentation nitpick

**Review by consideRatio — COMMENTED** [review]
With resolved comments, this LGTM

File: eksctl/carbonplan.jsonnet (hunk 2 above)
Comment (yuvipanda) [reply eksctl/carbonplan.jsonnet:78]: @consideRatio yeah, I was trying to be 'clever' and just be able to set `instanceType` for both non-spot and spot node pools. However, this becomes unmaintainable soon, so I've reverted that. Now the config is the same as what you'll pass to eksctl.

**Comment by yuvipanda** [comment]
Thanks a lot, @consideRatio! what do you think of it now?

**Review by consideRatio — APPROVED** [review]
LGTM!

I don't feel comfortable in this situation: either I press merge or I don't.

If I merge, I'm not feeling comfortable as I don't know what happens in the CI system and feel that I have a responsibility if I merge something and the CI system messes things up in a deployment or something happen making me think so.

If I don't press merge, I feel that it would slowed down a process that isn't worth slowing down after one approved review in this case.

I'll opt to not press merge this time and count on self-merge is to be considered an acceptable policy given this PR review approval.

**Comment by choldgraf** [comment]
@consideRatio we're discussing that topic over in here too: https://github.com/2i2c-org/team-compass/issues/175

**Pull request merged**